### PR TITLE
Format markdown

### LINKED ISSUE #48: Sort out formatting
Borrow the markdownlint and other formatting goodness from the main OpenAPI Specification to make this more readable. Remove the Table of Contents as well - it isn't needed since both ReSpec and GitHub will create them

### PR DESCRIPTION
**Pull request by ralfhandl**
Fixes #48

- [x] scripts/format-markdown.sh: bash script to reformat source(s)
- [x] package.json
  - [x] add dependencies
  - [x] add npm script `format-markdown`
- [x] package-lock.json: check in to fix dependencies
- [x] .markdownlint.yaml: added
- [x] versions/1.0.0.md: use respec-style normative references
- [x] build.sh: remove tweak code for normative references

Part of
* #53 

Partially conflicts with
* #47

### CODE REVIEW

File: scripts/format-markdown.sh
Comment (kevinswiber) [inline scripts/format-markdown.sh:1]: I know the package.json script runs this with `bash ./scripts/format-markdown.sh`, but if there's going to be a shebang (`#!/bin/bash`), we should probably `chmod +x ./scripts/format-markdown.sh` so it's executable.  (Or remove the shebang.)

File: scripts/md2html/build.sh [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+  version=$(basename "$filename" .md)
+  tempfile=../../deploy/overlay/v$version-tmp.html
+  echo -e "\n=== v$version ==="
+  node md2html.js --maintainers ./history/EDITORS_v$version.md ${filename} > $tempfile
+  npx respec --use-local --src $tempfile --out ../../deploy/overlay/v$version.html
+  rm $tempfile
+  if [ $version = $latest ]; then
+    if [[ ${version} != *"rc"* ]];then
+      # version is not a Release Candidate
+      cp -p ../../deploy/overlay/v$version.html ../../deploy/overlay/latest.html
```
Comment (kevinswiber) [inline scripts/md2html/build.sh:30]: I wish we could do a 307 Temporary Redirect here but happy we can at least add a canonical URL to the HTML.

Comment (ralfhandl) [reply scripts/md2html/build.sh:30]: This seems possible with https://github.com/jekyll/jekyll-redirect-from?tab=readme-ov-file#redirect-to.

Anyone feeling an irresistible urge to dive deeper into Jekyll?

**Comment by lornajane** [comment]
Looks good, but can you update now that #46 is merged so we can see which changes are here?

**Comment by ralfhandl** [comment]
Updated from `main`, now contains only formatting-related changes.

File: scripts/md2html/build.sh (hunk 1 above)
Comment (ralfhandl) [reply scripts/md2html/build.sh]: Created #54 to track this.

**Review by lornajane — APPROVED** [review]

**Pull request merged**